controllers/windowsmachine_controller.go

```diff
@@ -515,3 +507,17 @@ func (r *WindowsMachineReconciler) isWindowsMachineHealthy(machine *mapi.Machine
 
 	return true
 }
+
+// getInternalIPAddress returns the internal IP address of the Machine
+func getInternalIPAddress(addresses []core.NodeAddress) (string, error) {
+	// Get the IP address associated with the Windows machine, if not error out to requeue again
+	if len(addresses) == 0 {
+		return "", errors.New("no IP addresses defined")
+	}
+	for _, address := range addresses {
+		if address.Type == core.NodeInternalIP {
+			return address.Address, nil
+		}
+	}
+	return "", errors.Errorf("no internal IP address associated")
```

Comment:
can use errors.new here if there isn't a format string to be specified.

Reply:
Done.